### PR TITLE
rpc: add subscription id to events

### PR DESCRIPTION
**Pull request by tychoish**
Addresses  #3931

### CODE REVIEW

File: rpc/client/local/local.go [hunk 1]
```diff
@@ -285,11 +291,13 @@ func (c *Local) resubscribe(subscriber string, q tmpubsub.Query) types.Subscript
 }
 
 func (c *Local) Unsubscribe(ctx context.Context, subscriber, query string) error {
-	q, err := tmquery.New(query)
+	args := tmpubsub.UnsubscribeArgs{Subscriber: subscriber}
+	var err error
+	args.Query, err = tmquery.New(query)
 	if err != nil {
-		return fmt.Errorf("failed to parse query: %w", err)
+		args.ID = query
```
Comment (cmwaters) [inline rpc/client/local/local.go:298]: Is this / will this always be the correct way to handle this error?

**Review by cmwaters — COMMENTED** [review]
I've just added some comments in passing as I try to understand the changes. I'm assuming this is to solve the case in https://github.com/tendermint/tendermint/issues/3931

The issue I understand is that we want to generate id's so that clients can cancel individual subscriptions. AFAICS, the query itself serves currently as the unique identifier for subscribing and unsubscribing i.e. a single client can subscribe to multiple queries. From this, I assume this PR therefore targets the case when a single client wants multiple subscriptions to the same query. This appears to me a bit illogical - why would I want to have multiple subscriptions of the same query? Even if multiple services where using the same query on the client side can't they handle it? Maybe I've misunderstood something

**Review by cmwaters — APPROVED** [review]
This seems all in order to me. 

I still think that in the long run we should just track the subscriptions by id and not some hybrid of both.

**Comment by marbar3778** [comment]
is this ready to be merged?

**Pull request merged**